services/ble/Gatt.hpp

```diff
@@ -109,26 +110,39 @@ namespace services
         AttAttribute::Handle endHandle;
     };
 
-    class AttMtuExchange;
+    class AttMtuExchangeInterface;
 
     class AttMtuExchangeObserver
-        : public infra::Observer<AttMtuExchangeObserver, AttMtuExchange>
+        : public infra::Observer<AttMtuExchangeObserver, AttMtuExchangeInterface>
     {
     public:
-        using infra::Observer<AttMtuExchangeObserver, AttMtuExchange>::Observer;
+        using Observer::Observer;
 
         virtual void ExchangedMaxAttMtuSize() = 0;
     };
 
-    class AttMtuExchange
+    class AttMtuExchangeInterface
         : public infra::Subject<AttMtuExchangeObserver>
     {
+    protected:
+        ~AttMtuExchangeInterface() = default;
+
     public:
         virtual uint16_t EffectiveMaxAttMtuSize() const = 0;
```

Comment:
The name is double: max is already part of MTU
```suggestion
        virtual uint16_t EffectiveAttMtuSize() const = 0;
```